runelite-client/src/main/java/net/runelite/client/plugins/timetracking/farming/GeomancyManager.java

```diff
@@ -0,0 +1,238 @@
+			if (fp == null)
+			{
+				log.error("Couldn't find farming patch to match Jagex patch ID " + jagexPatchID);
+				return;
+			}
+
+			//Protection
+			boolean gardenerProtection = testBit(flags, 2);
+			boolean flowerProtection = testBit(flags, 3);
+			boolean hosidiusProtection = testBit(flags, 8);
+			boolean isProtected = gardenerProtection || flowerProtection || hosidiusProtection;
+			boolean wasProtected = paymentTracker.getProtectedState(fp);
+			if (wasProtected != isProtected)
+			{
+				isDirty = true;
+				paymentTracker.setProtectedState(fp, isProtected);
+			}
+
+			//Compost
+			CompostState compostState = null;
+			if (testBit(flags, 7))
+			{
+				compostState = CompostState.ULTRACOMPOST;
+			}
+			else if (testBit(flags, 6))
+			{
+				compostState = CompostState.SUPERCOMPOST;
+			}
+			else if (testBit(flags, 5))
+			{
+				compostState = CompostState.COMPOST;
+			}
+			CompostState previousCompostState = compostTracker.getCompostState(fp);
+			if (previousCompostState != compostState)
+			{
+				isDirty = true;
+				compostTracker.setCompostState(fp, compostState);
+			}
+
+			//Compost bins have very different states so let's not try to update them via Geomancy
+			if (fp.getImplementation() == PatchImplementation.COMPOST || fp.getImplementation() == PatchImplementation.GIANT_COMPOST)
+			{
+				return;
+			}
+
+			//Patch State
+			PatchState patchState = FarmingTracker.GeneratePatchState(cropItemID, data.currentStage, data.maxStage, flags);
+			String storedValue = configManager.getConfiguration(TimeTrackingConfig.CONFIG_GROUP, configManager.getRSProfileKey(), fp.configKey());
+			PatchState previousPatchState = null;
+			if (storedValue != null)
+			{
+				previousPatchState = farmingTracker.convertStoredValueToPatchStateAndTime(fp, storedValue).patchState;
+			}
+			//Don't overwrite stage when it was already harvestable since we don't get data on harvest stages via Geomancy
+			if (previousPatchState != null && previousPatchState.getCropState() == CropState.HARVESTABLE && patchState.getCropState() == CropState.HARVESTABLE)
+			{
+				patchState = new PatchState(patchState.getProduce(), patchState.getCropState(), previousPatchState.getStage());
+			}
+			if (previousPatchState == null || !previousPatchState.equals(patchState))
+			{
+				if (enableVerboseLogging)
+				{
+					log.info("Found difference, overwriting patch state.\nOld State: " + previousPatchState + " \n New State:" + patchState);
+				}
+				isDirty = true;
+				farmingTracker.loadCompletionTimes();
+				long unixNow = Instant.now().getEpochSecond();
+				String value = jagexPatchID + ":" + cropItemID + ":" + flags + ":" + coords + ":" + unixNow;
+				configManager.setRSProfileConfiguration(TimeTrackingConfig.CONFIG_GROUP, fp.configKey(), value);
+			}
+		}
+	}
+
+	private void LogPatchInfo(int jagexPatchID, int cropItemID, int minimumYield, int isWatered, int currentStage, int maxStage, int flags)
```

Comment:
delete this method

Reply:
addressed in 2ffdbad23653acde21afe94f03ff41af4c7af20a